src/string.cr

```diff
@@ -2385,46 +2385,45 @@ class String
     cmp == 0 ? (bytesize <=> other.bytesize) : cmp.sign
   end
 
-  # Compares this string with *other*, returning `-1`, `0` or `+1` depending on whether
-  # this string is less, equal or greater than *other*, optionally in a *case_insensitive*
-  # manner.
-  #
-  # If *case_insitive* is `false`, this method delegates to `<=>`. Otherwise,
-  # the strings are compared char-by-char, and ASCII characters are compared in a
-  # case-insensitive way.
+  # Returns `true` if `self` equals *other* in a case-insensitive manner.
   #
   # ```
-  # "abcdef".compare("abcde")   # => 1
-  # "abcdef".compare("abcdef")  # => 0
-  # "abcdef".compare("abcdefg") # => -1
-  # "abcdef".compare("ABCDEF")  # => 1
-  #
-  # "abcdef".compare("ABCDEF", case_insensitive: true) # => 0
-  # "abcdef".compare("ABCDEG", case_insensitive: true) # => -1
+  # "HellO".equals?("hELLo")             # => true
+  # "hello world".equals?("HELLO WORLD") # => true
+  # "Ü".equals?("ü")                     # => true
+  # "hello".equals?("world")             # => false
   # ```
-  def compare(other : String, case_insensitive = false)
-    return self <=> other unless case_insensitive
+  def equals?(other : String)
```

Comment:
Why is `equals?` case insensitive now? It's not intuitive, and also not configurable..